src/app/lib/models/generic_collection.js

```diff
@@ -8,11 +8,19 @@
         var idsPromise = torrentsPromise.then(_.bind(torrentProvider.extractIds, torrentProvider));
         var promises = [
             torrentsPromise,
-            subtitle ? idsPromise.then(_.bind(subtitle.fetch, subtitle)) : true,
+
+            subtitle ? idsPromise.then(_.bind(subtitle.fetch, subtitle)).catch(function (err) {
+                win.warn('Cannot fetch subtitles (%s):', torrentProvider.name, err);
```

Comment:
let's not use win.warn as they don't give us good stack traces, please use console.error